### DISCUSSION TITLE
Help with enhancing a short script

### DISCUSSION THREAD

**Discussion by cleanev** [teslamate-org/teslamate#2651] [Q&A] [answered]
@Dulanic - requesting your assistance if you can help me add a column to see difference between start and stop for sleep length.

Original script is posted by @BriCo94 at this link-https://github.com/adriankumpf/teslamate/issues/2556#issuecomment-1084034935

Is it possible to enhance it to do the math and display difference in mins/hours between start and stop time, please? Like so in below picture
![12388638-890C-4417-B948-56DCCC8D8101](https://user-images.githubusercontent.com/45519661/168184823-a2c606b1-abfe-4a81-8301-d1c4ba4697ed.jpeg)

Any assistance greatly appreciated.

**Comment by Dulanic** [comment, marked as answer]
It's easy to add, if you want to add it to grafana you just need to set the data type.

SQL just add this to the select query...

` ,extract(EPOCH FROM end_date - start_date ) as timediff`

This query will work:

```
SELECT *
    ,extract(EPOCH FROM end_date - start_date )
FROM states
where car_id =1 and (state ='asleep' or state = 'offline') and
extract(EPOCH FROM end_date - start_date ) >= 3600
ORDER BY 1
```

Then you just need to set the type to duration(s) which grafana will convert to a human readable format.

![image](https://user-images.githubusercontent.com/2948206/168314700-0c11c689-3f19-4421-9030-72c424ffdd80.png)

**Reply by cleanev** [reply]
Appreciate your help. Now I can see time difference, also saved new dashboard with desc added at the end to the SQL query to show newest ones at the top. I had to use ‘duration (d hh:mm:ss)’ for standard options -> Unit as my car just sleeps for 45-49 minutes at a time, otherwise it was showing in seconds. A lot to learn and thanks for the assistance.

Not sure how to make my car sleep for hours like yours (that’s a whole different topic)
